models/repo.go

```diff
@@ -1273,6 +1273,17 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// Create/Remove git-daemon-export-ok for git-daemon...
+		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
+		if repo.IsPrivate {
+			// NOTE: Gogs doesn't actually care about this file so we don't do any error-checking :D
+			os.Remove(daemonExportFile)
+		} else {
+			// NOTE: Gogs doesn't actually care about this file so we don't do any error-checking :D
+			f, _ := os.Create(daemonExportFile)
```

Comment:
Any reason not to use `os.Create`? they perform the same task, just that `os.Create` is self-documenting while `ioutil.WriteFile` 0 bytes isn't (IMO). Since what we're doing is _creating_ a file, not writing to it :smile: 

I'll fix the error-checking though. Might as well make the admin aware of the fact that something is wrong :laughing: